app/index.js

```diff
@@ -0,0 +1,104 @@
+const calculator = document.querySelector(".calculator");
+const keys = document.querySelector(".calculator-key");
+const display = document.querySelector(".calculator_display");
+
+keys.addEventListener("click", (e) => {
+  if (e.target.matches("button")) {
+    const key = e.target;
+    const action = key.dataset.action;
+    const keyContent = key.textContent;
+    const displayedNum = display.textContent;
+
+    let previousKeyType = calculator.dataset.previousKeyType;
+    Array.from(key.parentNode.children).forEach((k) =>
+      k.classList.remove("is-depressed"),
+    );
+
+    if (!action) {
+      if (
+        displayedNum === "0" ||
+        previousKeyType === "operator" ||
+        previousKeyType === "calculate"
+      ) {
+        display.textContent = keyContent;
+      } else {
+        display.textContent = displayedNum + keyContent;
+      }
+      calculator.dataset.previousKeyType = "number";
+    }
+
+    if (
+      action === "add" ||
+      action === "subtract" ||
+      action === "multiply" ||
+      action === "divide"
+    ) {
+      const firstValue = calculator.dataset.firstValue;
+      const operator = calculator.dataset.operator;
+      const secondValue = displayedNum;
+
+      if (
+        firstValue &&
+        operator &&
+        previousKeyType !== "operator" &&
+        previousKeyType !== "calculate"
+      ) {
+        const calValue = calculateFunc(firstValue, operator, secondValue);
+        display.textContent = calValue;
+        calculator.dataset.firstValue = calValue;
+      } else {
+        calculator.dataset.firstValue = displayedNum;
+      }
+
+      key.classList.add("is-depressed");
+      calculator.dataset.previousKeyType = "operator";
+      calculator.dataset.operator = action;
+    }
+
+    if (action === "decimal") {
+      if (!display.textContent.includes(".")) {
+        display.textContent = displayedNum + ".";
+      } else if (
+        previousKeyType === "operator" ||
+        previousKeyType === "calculate"
+      ) {
+        display.textContent = "0.";
+      }
+      calculator.dataset.previousKeyType = "decimal";
+    }
+
+    if (action !== "clear") {
+      const clearButton = document.querySelector("[data-action=clear]");
+      clearButton.textContent = "CE";
+    }
+
+    if (action === "clear") {
+      key.textContent = "AC";
+      calculator.dataset.firstValue = "";
+      calculator.dataset.modValue = "";
+      calculator.dataset.operator = "";
+      calculator.dataset.previousKeyType = "";
+
+      display.textContent = 0;
+      calculator.dataset.previousKeyType = "clear";
+    }
+
+    if (action === "calculate") {
```

Comment:
This existing code has some deep nesting. We can discuss the challenges this creates and the solutions available as part of a new effort later this week.